OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ApplicationController.cs

```diff
@@ -167,14 +168,74 @@ public async Task<IActionResult> GetByProviderId(Guid providerId, [FromQuery] Ap
 
         var applications = await applicationService.GetAllByProvider(providerId, filter).ConfigureAwait(false);
 
-        if (applications == null || !applications.Entities.Any())
+        if (CheckApplicationsFor204(applications))
         {
             return NoContent();
         }
 
         return Ok(applications);
     }
 
+    /// <summary>
+    /// Get collection of applications, that have a pending status.
+    /// </summary>
+    /// <param name="providerId">Provider id.</param>
+    /// <returns>List of applications.</returns>
+    /// <response code="200">Entities were found by given Id.</response>
+    /// <response code="204">No entity with given Id was found.</response>
+    /// <response code="500">If any server error occurs.</response>
+    [HasPermission(Permissions.ApplicationRead)]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(SearchResult<ApplicationDto>))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpGet("/api/v{version:apiVersion}/provider/{providerId}/applications/pending")]
+    public async Task<IActionResult> GetPendingApplicationsByProviderId(Guid providerId)
+    {
+        // Find a provider with given id in admins and standard providers
+        var providerStandard = await providerService.GetById(providerId).ConfigureAwait(false);
+
+        var providerAdminIdStringVersion = providerId.ToString();
+        var providerAdmin = await providerAdminService.GetById(providerAdminIdStringVersion).ConfigureAwait(false);
+
+        // If any provider was not found - return BadRequest
+        if (providerStandard is null && providerAdmin is null)
+        {
+            return BadRequest($"There is no any provider with Id = {providerId}");
+        }
+
+        // Filter for applications, that is set to "Pending"
+        var filter = new ApplicationFilter()
+        {
+            Statuses = new List<ApplicationStatus>()
+            {
+                ApplicationStatus.Pending,
+            },
+        };
+
+        SearchResult<ApplicationDto> applications = null;
+
+        // Get applications from not null provider
+        if (providerStandard is not null)
+        {
+            applications = await applicationService.GetAllByProvider(providerId, filter).ConfigureAwait(false);
+        }
+        else
+        {
+            applications = await applicationService
+                                     .GetAllByProviderAdmin(providerAdminIdStringVersion, filter, providerAdmin.ProviderId, providerAdmin.IsDeputy)
+                                     .ConfigureAwait(false);
+        }
+
+        // If applications is null or empty - return 204
+        if (CheckApplicationsFor204(applications))
+        {
+            return NoContent();
+        }
+
+        return Ok(applications.TotalAmount);
```

Comment:
(Logic) Action name and comment promise to return applications. Instead it returns count. Please fix something.

Reply:
Now returns collection. Fixed.